### DISCUSSION TITLE
how to delivery data from same Producer Factory with different params

### DISCUSSION THREAD

**Discussion by conderls** [SETL-Framework/setl#263] [Q&A]
If I have a producer Factory whose result depends on param `label`,
```scala
class ProducerFactory(label: String) extends Factory[String] {
   override def deliveryId = "delivery_id"
   ...
  override def get() = label
}
```

I may use the data from ProducerFactory in another Factory by 
```scala
class ConsumerFactory(label: String) extends Factory[String] {
   @Delivery(id = "delivery_id")
   private val label: String = ""
   ...
}
```

I can override `deliveryId` in ProducerFactory according to the param `label`, e.g. `override def deliveryId = s"delivery_id_$label"` for sure.

1. I am wondering that, the ConsumerFactory is generic enough, If I have Producers with `deliveryId`: `"delivery_id_a", "delivery_id_b"`, how can I using ConsumerFactory to consume data from these Producers? As I cannot set deliveryId in annotation according to the param `label`
```scala
   @Delivery(id = s"delivery_id_$label")
   private val label: String = "" // ERROR raised: annotation argument needs to be a constant;
```

2. right now, I am creating new pipeline to use ProducerFactory and ConsumerFactory with fixed `deliveryId`; but how to clean the pipeline if there are some cached/persisted data.

**Comment by glc-mfelja** [comment]
duplicated #274
